src/clover/tests/integration/test_clover.py

```diff
@@ -0,0 +1,967 @@
+
+        The test simulation uses:
+            - a PV system size of 20 kWP;
+            - a storage system size of 25 kWh.
+
+        """
+
+        info_file_data = self._run_clover_simulation(
+            True, True, True, True, pv_size=20, storage_size=25
+        )
+        self._check_output(
+            info_file_data,
+            average_daily_diesel=1.65,
+            average_daily_grid_energy=7.196,
+            average_daily_grid_times=9.338,
+            average_daily_renewables_energy=4.262,
+            average_daily_storage_energy=7.52,
+            blackouts=0.1,
+            cumulative_cost=32249.869,
+            cumulative_ghgs=91620.383,
+            cumulative_pv_generation=36685.0,
+            diesel_capacity=3.0,
+            diesel_times=0.12,
+            final_pv_size=19.0,
+            final_storage_size=21.34,
+            initial_pv_size=20.0,
+            initial_storage_size=25.0,
+            lcue=1.124,
+            renewables_fraction=0.571,
+            unmet_energy_fraction=0.018,
+        )
+
+    @pytest.mark.integrest
+    def test_diesel_grid_and_pv(self):
+        """
+        Tests the case with diesel, grid and PV.
+
+        The test simulation uses:
+            - a PV system size of 20 kWP.
+
+        """
+
+        info_file_data = self._run_clover_simulation(
+            True, True, True, False, pv_size=20
+        )
+        self._check_output(
+            info_file_data,
+            average_daily_diesel=8.821,
+            average_daily_grid_energy=7.196,
+            average_daily_grid_times=9.338,
+            average_daily_renewables_energy=4.262,
+            average_daily_storage_energy=0.0,
+            blackouts=0.1,
+            cumulative_cost=26343.181,
+            cumulative_ghgs=100650.761,
+            cumulative_pv_generation=36685.0,
+            diesel_capacity=3.0,
+            diesel_times=0.394,
+            final_pv_size=19.0,
+            final_storage_size=0.0,
+            initial_pv_size=20.0,
+            initial_storage_size=0.0,
+            lcue=0.934,
+            renewables_fraction=0.21,
+            unmet_energy_fraction=0.016,
+        )
+
+    @pytest.mark.integrest
+    def test_diesel_grid_and_storage(self):
+        """
+        Tests the case with diesel, grid and storage.
+
+        The test simulation uses:
+            - a storage system size of 25 kWh.
+
+        """
+
+        info_file_data = self._run_clover_simulation(
+            True, True, False, True, storage_size=25
+        )
+        self._check_output(
+            info_file_data,
+            average_daily_diesel=11.149,
+            average_daily_grid_energy=7.196,
+            average_daily_grid_times=9.338,
+            average_daily_renewables_energy=0.0,
+            average_daily_storage_energy=0.005,
+            blackouts=0.099,
+            cumulative_cost=15841.411,
+            cumulative_ghgs=42714.989,
+            cumulative_pv_generation=0.0,
```

Comment:
This is the first place where I'd have expected CLOVER to output that no electricity was generated from the PV panels, but, it's actually outputting that a large amount of PV generation happened, as it's simply summing up the renewables.ninja data, rather than involving anything to do with the scenario etc.

Reply:
The test is looking at the outputs of clover in the `info_data.json` file and CLOVER is outputting the information there 😱

Reply:
I've now updated this in discussion with @phil-sandwell 🎉